### PR TITLE
Removed use of lock(this), removed dead code

### PR DESCRIPTION
**Pull request by snakefoot**
Removed use of lock(this) to make code analysis happy

### CODE REVIEW

File: src/NLog/Config/XmlLoggingConfiguration.cs [hunk 1]
```diff
@@ -470,20 +470,20 @@ private void ParseNLogElement(ILoggingConfigurationElement nlogElement, string f
         /// <summary>
         /// Parses a single config section within the NLog-config
         /// </summary>
-        /// <param name="section"></param>
+        /// <param name="configSection"></param>
         /// <returns>Section was recognized</returns>
-        protected override bool ParseNLogSection(ILoggingConfigurationElement section)
+        protected override bool ParseNLogSection(ILoggingConfigurationElement configSection)
```
Comment (304NotModified) [inline src/NLog/Config/XmlLoggingConfiguration.cs]: unfortunately this is a binary breaking change

File: src/NLog/Targets/DefaultJsonSerializer.cs [hunk 2]
```diff
@@ -432,26 +432,6 @@ private void SerializeCollectionObject(IEnumerable value, StringBuilder destinat
             destination.Append(']');
         }
 
-        private bool SerializeTypeCodeValue(object value, TypeCode objTypeCode, StringBuilder destination, JsonSerializeOptions options, SingleItemOptimizedHashSet<object> objectsInPath, int depth)
```
Comment (304NotModified) [inline src/NLog/Targets/DefaultJsonSerializer.cs]: this is not needed anymore? Please do describe these changes in the PR title to make the review less of a surprise - thanks!

Comment (304NotModified) [reply src/NLog/Targets/DefaultJsonSerializer.cs]: I will update it for now

File: src/NLog/Config/XmlLoggingConfiguration.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/Config/XmlLoggingConfiguration.cs]: Pretty sure we have not released NLog 4.6 yet

File: src/NLog/Targets/DefaultJsonSerializer.cs (hunk 2 above)
Comment (snakefoot) [reply src/NLog/Targets/DefaultJsonSerializer.cs]: Thx

File: src/NLog/Config/XmlLoggingConfiguration.cs (hunk 1 above)
Comment (304NotModified) [reply src/NLog/Config/XmlLoggingConfiguration.cs]: ah yes, then it's OK indeed :)

**Pull request merged**